### PR TITLE
Pullrequest apple push notifications

### PR DESCRIPTION
**Pull request by BenSartor**
Hi @jritzerfeld,

here the weird things come in :)
Furthermore this pr has some rough edges which I would like to discuss with you.

Never the less it would be realy great, if you pay a special attention at the apple ssl certificate pinnging.

Thanks a lot
 Ben

### CODE REVIEW

File: src/test/java/org/simlar/simlarserver/services/pushnotification/ApplePushNotificationTest.java [hunk 1]
```diff
@@ -96,4 +98,13 @@ public void testConnectToAppleWithCertificateBadDeviceToken() {
             assertEquals("{\"reason\":\"BadDeviceToken\"}", e.getResponseBodyAsString());
         }
     }
+
+    @SuppressFBWarnings("UTAO_JUNIT_ASSERTION_ODDITIES_NO_ASSERT")
+    @SuppressWarnings({"JUnitTestMethodWithNoAssertions", "PMD.JUnitTestsShouldIncludeAssert"})
```
[the remark this answers is not in the document]
Comment (BenSartor) [reply src/test/java/org/simlar/simlarserver/services/pushnotification/ApplePushNotificationTest.java]: Yes, and every static code analyzer we use complains about it.
Would you suggest adding a return value to requestVoipPushNotification()?
The Apple server does not return anything just status code 200.

[the remark this answers is not in the document]
Comment (jritzerfeld) [reply src/test/java/org/simlar/simlarserver/services/pushnotification/ApplePushNotificationTest.java]: Really? [Handling Notification Responses from APNs](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/handling_notification_responses_from_apns)

[the remark this answers is not in the document]
Comment (BenSartor) [reply src/test/java/org/simlar/simlarserver/services/pushnotification/ApplePushNotificationTest.java]: Good point. I did not see the apns-id in the header.
We could return it, although we do not use it. Maybe logging.
Would you prefer that?

[the remark this answers is not in the document]
Comment (jritzerfeld) [reply src/test/java/org/simlar/simlarserver/services/pushnotification/ApplePushNotificationTest.java]: Yes, please. :hugs:

**Comment by BenSartor** [comment]
Hi @jritzerfeld,
do we have to todos left before merging this to the push-notification branch?
As always thanks a lot and cheers,
 Ben

**Pull request merged**